invenio_accounts/api.py

```diff
@@ -8,25 +8,60 @@
 
 """API objects for Invenio Accounts."""
 
-from collections import defaultdict
+
+class Session:
+    """Session object for DB Users change history."""
+
+    def __init__(self):
+        self.updated_users = []
+        self.updated_roles = []
+        self.deleted_users = []
+        self.deleted_roles = []
+        self.indexed = False
+        self.invalidated_cache = False
 
 
 class DBUsersChangeHistory:
     """DB Users change history storage."""
 
     def __init__(self):
-        """constructor."""
-        # the keys are going to be the sessions, the values are going to be
-        # the sets of dirty/deleted models
-        self.updated_users = defaultdict(lambda: list())
-        self.updated_roles = defaultdict(lambda: list())
-        self.deleted_users = defaultdict(lambda: list())
-        self.deleted_roles = defaultdict(lambda: list())
-
-    def _clear_dirty_sets(self, session):
-        """Clear the dirty sets for the given session."""
+        """Constructor."""
+        self.sessions = {}
+
+    def _get_session(self, session_id):
+        if session_id not in self.sessions:
+            self.sessions[session_id] = Session()
+        return self.sessions[session_id]
+
+    def add_updated_user(self, session_id, user_id):
+        session = self._get_session(session_id)
+        if user_id not in session.updated_users:
```

Comment:
if we change the objects from list to sets we do not need to perform a check, since the set contains unique values. So we would just call `add`.

```python
>>> test = set()
>>> test.add(1)
>>> test
{1}
>>> test.add(1)
>>> test
{1}
```

The same applies in other functions.